### ISSUE TITLE
[PC ENGINE CD] General Game Audio Issues

### ISSUE THREAD

**Issue by boktai1000** [higan-emu/ares#433]
Version tested - Ares v127 (but also previously tested on v120 and v126)

I can start to add more information here but wanted to get this down. Testing various PC Engine CD games such as Ys IV - The Dawn of Ys (English Text + Dub) and Akumajou Dracula X - Chi No Rondo (English v1.01) both have audio issues of differing degrees. In the example of Ys, once you start the game not shortly after a bit of the introduction of the game, one of the characters dialogue simply cuts out. With Dracula X (Rondo of Blood) once you start fighting Death in the beginning of the game, you'll note that various sound effects simply do not play correctly, while the background track continues to play.

There are probably lots more examples here, but I've been able to get this to replicate with these two games for example with various versions of Ares listed above.

Let me know if more information is needed to help troubleshoot, since it was audio related - screenshots didn't seem to make sense unless its to capture when the audio issues are first noticeable, but a description felt sufficient. I can add them later though if necessary.

**Comment by boktai1000** [comment]
Issues with Akumajou Dracula X - Chi No Rondo (English v1.01) become noticeable at this section of the game when it becomes playable (just after the intro video, about 2 minutes into starting a file if you include the intro video)

![image](https://user-images.githubusercontent.com/9054563/157259367-535d17f2-31bc-4efb-80bb-927a443d2adf.png)

Issues with Ys IV - The Dawn of Ys (English Text + Dub) occur right at the beginning of the game during this part of the opening of the game after they say "Hey Adol, look". There also seems to be a bit of popping that occurs before/after dialogue starts, and some dialogue simply gets cut out - letting the intro play out without interaction you should notice these issues and miss lots of dialogue as a result since its spoken.

![image](https://user-images.githubusercontent.com/9054563/157259896-faa46dc1-33ac-4833-9e49-9ec34e7daf02.png)

Issues with Tokimeki Memorial (Japan) (Rev 3) appear with the opening where the audio is delayed and doesn't match the lyrics on the screen, it appears to be slightly off. This is compared to other emulators where it appears on the screen as it is spoken. A screenshot showing the part of the opening is included below. This copy is Redump verified and not a patched version as one does not exist, so all track files are indeed present.

![image](https://user-images.githubusercontent.com/9054563/157261693-0c2fd6d9-5caf-4998-8628-344fbbadab59.png)

I'm sure there a plethora of other audio related issues, and I could flag them as I come across them - but if it's superfluous I don't want to create too much noise - just enough to help possibly pinpoint finding the issue. I hope this helps track down the root cause, and if more information is needed on my end just reply and I'll follow up when I can!

**Comment by HeroponRikiBestest** [comment]
Another audio issue is present in Babel (Japan). Normally, the game will pause the redbook audio track playing at a given time, seemingly because it needs to load certain sprites/portraits/etc in. Once they're loaded, the redbook audio track will resume. However, ares will never resume the redbook track, leaving no music playing.

Also, basically every game I've tried has a desync between the audio and the visuals, especially noticeable with lip syncing. Are there any you've found that don't have this issue?

**Comment by boktai1000** [comment]
@HeroponRikiBestest so for full disclosure and context - Ares is (was) my main emulator that I used for SNES, NES, Game Boy, etc. I've recently gotten into PC Engine and was delighted to see that Ares supported it natively. The first game that I tried to play in Ares was Rondo of Blood, which is when I immediately noticed these audio issues only minutes after booting it up.

Personally speaking, I consider PC Engine `Experimental` or borderline Unplayable given the context of the game and audio in it's current state with these current audio issues, I don't want to come off too harsh but is the reality of the current support. The website https://ares-emu.net/ lists PC Engine as supported but I would consider these audio related issues with nearly ever title I've thrown at it to qualify as `Experimental` instead, which is a tag given to PlayStation, etc.

I've since looked to use a different emulator (BizHawk) for PC Engine while Ares looks to resolve this issue. I would love to ultimately use Ares for PC Engine as well as assist/gather more information however I can that would help with troubleshooting and resolving this issue ultimately, as I think the Ares emulator is an excellent piece of software and what I want to use to play my games.

**Comment by LukeUsher** [comment]
> Personally speaking, I consider PC Engine Experimental

I think it's worth noting that the reason the PC-Engine core isn't marked as experimental is that the majority of the game library for the base system and supergrafx games are fine.

Most (all?) of these issues stem from incomplete emulation of the cd-rom drive; just like with how the Mega Drive core is pretty solid, but 32X games aren't well supported at this time.

**Comment by boktai1000** [comment]
You are correct, I misspoke and should have said PC Engine CD instead of PC Engine.
